sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala

```diff
@@ -577,4 +579,23 @@ class OrcQuerySuite extends QueryTest with BeforeAndAfterAll with OrcTest {
       assert(spark.table(tableName).schema == schema.copy(fields = expectedFields))
     }
   }
+
+  test("Empty schema does not read data from ORC file") {
+    val data = (1 to 2).zip(1 to 2)
+    withOrcFile(data) { path =>
+      val requrestedSchema = StructType(Nil)
+      val conf = new Configuration()
+      val physicalSchema = OrcFileOperator.readSchema(Seq(path), Some(conf)).get
+      OrcRelation.setRequiredColumns(conf, physicalSchema, requrestedSchema)
+      val maybeOrcReader = OrcFileOperator.getFileReader(path, Some(conf))
+      assert(maybeOrcReader.isDefined)
+      val orcRecordReader = new SparkOrcNewRecordReader(
+        maybeOrcReader.get, conf, 0, maybeOrcReader.get.getContentLength)
+
+      val recordsIterator = new RecordReaderIterator[OrcStruct](orcRecordReader)
+      assert(recordsIterator.next().toString == "{null, null}")
+
+      recordsIterator.close()
```

Comment:
Maybe, we should put some sugar on this something like `try {..} finally {recordsIterator.close()}`. I just wanted to show that it might be able to be tested :). if this test looks good to other guys as well.